src/pipe_log.rs

```diff
@@ -74,13 +79,88 @@ impl FileBlockHandle {
     }
 }
 
+/// Version of log file format.
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Eq,
+    PartialEq,
+    FromPrimitive,
+    ToPrimitive,
+    Serialize_repr,
+    Deserialize_repr,
+    EnumIter,
+)]
+#[repr(u64)]
+pub enum Version {
+    V1 = 1,
+    V2 = 2,
+}
+
+impl Version {
+    pub fn has_log_signing(&self) -> bool {
+        fail_point!("pipe_log::version::force_enable", |_| { true });
+        match self {
+            Version::V1 => false,
+            Version::V2 => true,
+        }
+    }
+}
+
+impl Default for Version {
+    fn default() -> Self {
+        Version::V1
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct LogFileContext {
+    pub id: FileId,
+    pub version: Version,
+}
+
+impl LogFileContext {
+    pub fn new(file_id: FileId, version: Version) -> Self {
+        Self {
+            id: file_id,
+            version,
+        }
+    }
+
+    /// Return the `signature` in `Option<u32>` format.
+    ///
+    /// `None` will be returned only if `self.version` is invalid.
+    pub fn get_signature(&self) -> Option<u32> {
+        fail_point!("pipe_log::version::force_get_signature", |_| {
+            match self.version {
```

Comment:
What's this failpoint for? It does the same thing as normal logic.

Reply:
It's an extra failpoint as a direct mode for getting the signature, ignoring the `has_log_signing` result. With it, it can make the `Version::V1` return None as expected.

Reply:
You can scope the failpoint like this:
```
let engine = {
  let _f = FailGuard::new("pipe_log::version::force_enable");
  Engine::open().unwrap()
};
engine.write(..);
```